homeassistant/components/camera/__init__.py

```diff
@@ -411,7 +426,19 @@ class CameraMjpegStream(CameraView):
     url = '/api/camera_proxy_stream/{entity_id}'
     name = 'api:camera:stream'
 
-    @asyncio.coroutine
-    def handle(self, request, camera):
-        """Serve camera image."""
-        yield from camera.handle_async_mjpeg_stream(request)
+    async def handle(self, request, camera):
+        """Serve camera stream, possibly with interval."""
+        interval = request.query.get('interval')
+        if interval is None:
+            await camera.handle_async_mjpeg_stream(request)
+            return
+
+        try:
+            # Compose camera stream from stills
+            interval = float(request.query.get('interval'))
+            await camera.handle_async_still_stream(request, interval)
+            return
+        except (ValueError, HomeAssistantError):
```

Comment:
When do we throw HomeAssistantError ?

Reply:
It's inside `handle_async_still_stream(request, interval)` on a too low value of interval. But that should probably also just be a ValueError?

Reply:
Yeah